beacon_node/network/src/network_beacon_processor/gossip_methods.rs

```diff
@@ -1044,7 +1017,7 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
                             "block_root" => %block_root,
                         );
 
-                        // Potentially trigger reconstruction
+                        self.attempt_data_column_reconstruction(block_root).await;
```

Comment:
Looks like this doesn't work, issues:

1. Still seeing reconstruction blocking gossip data column processing
2. This change should be made to the cached block instead of the clone: https://github.com/sigp/lighthouse/blob/7d2d826a892f46dd965484a1ecc49401db57ba54/beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs#L885
3. Because of #1, after the block is imported, I see a bunch of:
```
Jul 01 06:27:21.130 DEBG Attempted to publish duplicate message  kind: data_column_sidecar_31, service: libp2p
```

Reply:
Fixed #2.

I can't seem to figure out why reconstruction is blocking the processing of other incoming data columns. I don't see any lock being held during reconstruction, perhaps it is just waiting for a beacon processor worker to become available.

We could send it to the back of the beacon processor queue, but I'll do some more testing to make sure this is the case.